packages/html/ds/src/tabs/tabs.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  BaseComponent,
+  BaseComponentOptions,
+  initialiseModule,
+} from '../common/component';
+
+export type TabsOptions = BaseComponentOptions;
+
+class TabsAutomatic {
+  tabs: HTMLInputElement[];
+  firstTab!: HTMLInputElement;
+  lastTab!: HTMLInputElement;
+  tabElement: Element;
+  tabPanels: HTMLElement[];
+
+  constructor(tabElement: Element) {
+    this.tabElement = tabElement;
+
+    this.tabs = [];
+
+    this.tabs = [
+      ...this.tabElement.querySelectorAll('[role=tab]'),
```

Comment:
Can we tighten the selector on this one ? something like selecting it by class and role, I'm just thinking that if there are other elements with role of `tab` on the page, it will have unexpected behaviour.

Reply:
That `querySelectorAll` does queries within the scope of `tabElement`, so it does not reference any other tabs.